htlcswitch/link_test.go

```diff
@@ -6894,3 +6895,220 @@ func TestChannelLinkShortFailureRelay(t *testing.T) {
 	default:
 	}
 }
+
+func TestLinkFlushApiDirectionIsolation(t *testing.T) {
+	aliceLink, _, _, _, _, _ :=
+		newSingleLinkTestHarness(
+			t, 5*btcutil.SatoshiPerBitcoin,
+			1*btcutil.SatoshiPerBitcoin,
+		)
+
+	for i := 0; i < 10; i++ {
+		if prand.Uint64()%2 == 0 {
+			//nolint:errcheck
+			aliceLink.EnableAdds(Outgoing)
+		} else {
+			//nolint:errcheck
+			aliceLink.DisableAdds(Outgoing)
+		}
+		require.False(t, aliceLink.IsFlushing(Incoming))
+	}
+
+	//nolint:errcheck
```

Comment:
No because if it errors we still want to keep going. The "error" case here is just the system telling us that there was a noop state change. Sometimes we care about that, sometimes we don't. Here we don't.